### PR TITLE
Create login/registration screen

### PR DESCRIPTION
**Pull request by DekataBG**
Create login and registration screen, create repositories for communication with Firebase

### CODE REVIEW

File: DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    func signIn() {
+        Task {
+            do {
+                try await authenticationProvider.signIn(emailAddress: email, password: password)
+
+                appRootManager.currentRoot = .main
+            } catch {
+                print(error)
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift]: I think we can handle these errors that occur with a popup, and since we already have this functionality from the iOS standard library, I think it will be pretty easy to use it.

File: DigitalLibrary/DigitalLibrary/Models/Book.swift [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
     let description: String
     let author: String
     let publisher: String
     let year: Int
-    let qrcode: String? = nil // don't know how to configure it, so for it will be nil
-    let isbn: String? = nil // don't know how to configure it, so for it will be nil
-    let photo: String? = nil // don't know how to configure it, so for it will be nil
+//    let qrcode: String? // don't know how to configure it, so for it will be nil
+//    let isbn: String? // don't know how to configure it, so for it will be nil
+//    let photo: String? // don't know how to configure it, so for it will be nil
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/Models/Book.swift:18]: Why this is commented?

File: DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift (hunk 1 above)
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift]: Do you mean to create a popup with value OK and title the error title?

File: DigitalLibrary/DigitalLibrary/Models/Book.swift (hunk 2 above)
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/Models/Book.swift:18]: These were some properties that we haven't yet clarified how to handle. They are there to keep them in mind

File: DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift (hunk 1 above)
Comment (vasilkadt) [reply DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift]: Yeah, I mean we do not want to have print statements and it is good practise to have error handling. And to be more user friendly I think the user will want to know when the error happens and I think something like popup with error title and buttons like Dismiss for example or Try again and Cancel. You may decide which is more appropriate.

File: DigitalLibrary/DigitalLibrary/MainScreens/SignUpScreen/SignUpViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+            }
+        }
+    }
+
+    @Published var firstName: String = ""
+    @Published var lastName: String = ""
+    @Published var phoneNumber: String = ""
+    @Published var email: String = ""
+    @Published var password: String = ""
+    @Published var confirmPassword: String = ""
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/SignUpScreen/SignUpViewModel.swift:82]: I think they should be private

File: DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift (hunk 1 above)
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift]: I think just OK is the most appropriate. Try again is not appropriate for some scenarios. E.g. if you enter invalid credentials and you try again, you would always get the same error. Cancel is also not what the 
user actually does
Screenshot 2024-02-12 at 9 38 26

File: DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift
Comment (DesislavaIT) [inline DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift:1]: why is this file deleted?

File: DigitalLibrary/DigitalLibrary/Models/Book.swift (hunk 2 above)
Comment (DesislavaIT) [inline DigitalLibrary/DigitalLibrary/Models/Book.swift:18]: why?

File: DigitalLibrary/DigitalLibrary/Models/User.swift [hunk 4]
```diff
@@ -7,11 +7,13 @@
...
-    var books: [Book] = []
+struct UserModel: Codable {
+    let id: String
+    let firstName: String
+    let lastName: String
+    let email: String
+    let phone: String
+    var readingBooks: [Book]? = []
+    var droppedBooks: [Book]? = []
+    var readBooks: [Book]? = []
```
Comment (DesislavaIT) [inline DigitalLibrary/DigitalLibrary/Models/User.swift]: what droppedBooks mean?

File: DigitalLibrary/DigitalLibrary/MockData/BooksMockData.swift
Comment (DesislavaIT) [inline DigitalLibrary/DigitalLibrary/MockData/BooksMockData.swift:1]: why is this file deleted?

File: DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift:1]: I dont use it

File: DigitalLibrary/DigitalLibrary/Models/Book.swift (hunk 2 above)
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/Models/Book.swift:18]: We havent yet discussed how to handle these properties so I decided to comment them out

File: DigitalLibrary/DigitalLibrary/Models/User.swift (hunk 4 above)
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/Models/User.swift]: You read a book and at the 50th page you decide you dont like it and drop it. It is going in this category

File: DigitalLibrary/DigitalLibrary/MockData/BooksMockData.swift
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/MockData/BooksMockData.swift:1]: I dont use it

File: DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift
Comment (DesislavaIT) [reply DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift:1]: but I use it, revert this change

File: DigitalLibrary/DigitalLibrary/MockData/BooksMockData.swift
Comment (DesislavaIT) [reply DigitalLibrary/DigitalLibrary/MockData/BooksMockData.swift:1]: but I use it, revert this change

File: DigitalLibrary/DigitalLibrary/Models/User.swift (hunk 4 above)
Comment (DesislavaIT) [reply DigitalLibrary/DigitalLibrary/Models/User.swift]: unfinishedBooks is more understandable, please change the name

File: DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift
Comment (DekataBG) [reply DigitalLibrary/DigitalLibrary/MockData/UserMockData.swift:1]: ok
